easybuild/easyblocks/generic/bundle.py

```diff
@@ -308,3 +332,12 @@ def sanity_check_step(self, *args, **kwargs):
             fake_mod_data = self.load_fake_module(purge=True)
             self.log.debug("Cleaning up after testing loading of module")
             self.clean_up_fake_module(fake_mod_data)
+
+        if self.cfg['sanity_check_component']:
+            # run sanity checks for specific components
+            for idx, comp in enumerate(self.comp_cfgs_sanity_check):
+                print_msg("sanity checking bundle component %s v%s (%i/%i)...", comp.cfg['name'], comp.cfg['version'],
+                          idx + 1, len(self.cfg['sanity_check_component']))
+                self.log.info("Starting sanity check step for component %s v%s", comp.cfg['name'], comp.cfg['version'])
+
+                comp.run_step('sanity_check', [lambda x: x.sanity_check_step])
```

Comment:
No need to guard this, just loop over `self.comp_cfgs_sanity_check` (if no components are being sanity checked, it'll be an empty list).
You can also avoid the long lines by introducing local variables:

```suggestion
        # run sanity checks for specific components
        cnt = len(self.comp_cfgs_sanity_check)
        for idx, comp in enumerate(self.comp_cfgs_sanity_check):
            comp_name, comp_ver = comp.cfg['name'], comp.cfg['version']
            print_msg("sanity checking bundle component %s v%s (%i/%i)...", comp_name, comp_ver, idx + 1, cnt)
            self.log.info("Starting sanity check step for component %s v%s", comp_name, comp_ver)

            comp.run_step('sanity_check', [lambda x: x.sanity_check_step])
```